scipy/spatial/transform/rotation.py

```diff
@@ -281,3 +282,147 @@ def as_rotvec(self):
+            {'x', 'y', 'z'} or {'X', 'Y', 'Z'}. Each lowercase character
+            represents an extrinsic rotation around the corresponding axis and
+            an uppercase character represents an intrinsic rotation. Extrinsic
+            and intrinsic rotations cannot be mixed in one function call.
+        angles : float or array_like, shape (N, ) or (N, [1 or 2 or 3])
+            Euler angles specified in radians (`degrees` is False) or degrees
+            if parameter `degrees` is True.
+            For a single character `seq`, `angles` can be:
+
+                - a single value
+                - array_like, 1D with shape (N, ), where each `angle[i]`
+                  corresponds to a single rotation
+                - array_like, 2D with shape (N, 1), where each `angle[i, 0]`
+                  corresponds to a single rotation
+
+            For 2- and 3-character wide `seq`, `angles` can be:
+
+                - array_like, 1D with shape (W, ) where `W` is the width of
+                  `seq`, which corresponds to a single rotation with `W` axes
+                - array_like, 2D with shape (N, W) with each `angle[i]`
+                  corresponding to a sequence of Euler angles describing a
+                  single rotation
+
+        degrees : boolean
+            If True, then the given angles are taken to be in degrees
+        """
+
+        if len(seq) < 1 or len(seq) > 3:
+            raise ValueError("Expected axis specification to be a non-empty "
+                             "string of upto 3 characters, got {}".format(seq))
+
+        if ((len(seq) == 2 and seq[0] == seq[1]) or
+                (len(seq) == 3 and (seq[0] == seq[1] or seq[1] == seq[2]))):
+            raise ValueError("Expected consecutive axes to be different, "
+                             "got {}".format(seq))
+
+        intrinsic = (recompile('^[XYZ]{1,3}$').match(seq) is not None)
+        extrinsic = (recompile('^[xyz]{1,3}$').match(seq) is not None)
+        if not (intrinsic or extrinsic):
+            raise ValueError("Expected either intrinsic or extrinsic rotations"
+                             " , got {}".format(seq))
+
+        angles = np.asarray(angles, dtype=float)
+        if degrees:
+            angles = np.deg2rad(angles)
+
+        is_single = False
+        if len(seq) == 1:
+            # For single axis, intrinsic and extrinsic rotations are same
+            seq = seq.lower()
+            if angles.ndim == 0:
+                # Single value case
+                quat = cls._make_elementary_quat(seq, angles.reshape((1)))
+                is_single = True
+            elif angles.ndim == 1:
+                quat = cls._make_elementary_quat(seq, angles)
+            else:
+                raise ValueError("Expected 1D or 2D `angles` parameter for a"
+                                 "single character `seq`, "
+                                 "got shape {}".format(angles.shape))
+        else:  # 2 or 3 axes
```

Comment:
I think it's better to combine both cases into one. To do that we need to add extra dimensions to `angles` in case of one axis if necessary. I think it will make code easier to understand.

So the algorithms would look like this:

1. Prepare `angles` to have shape `(N, n_axes)`.
2. Call the `compose_elementary_rotations`.